client/pkg/omnictl/talosconfig.go

```diff
@@ -108,6 +112,13 @@ func getTalosconfig(args []string) func(ctx context.Context, client *client.Clie
 		if talosconfigCmdFlags.cluster == "" {
 			data, err = client.Management().Talosconfig(ctx, opts...)
 		} else {
+			_, err = client.Omni().State().Get(ctx, resource.NewMetadata(resources.DefaultNamespace, omni.ClusterType, talosconfigCmdFlags.cluster, resource.VersionUndefined))
+			if state.IsNotFoundError(err) {
+				fmt.Printf(`warning: cluster "%s" does not exist%s`, talosconfigCmdFlags.cluster, "\n")
+			} else if err != nil {
+				return err
+			}
```

Comment:
LGTM in general, but I'd suggest to change a couple of things here:

```suggestion
			_, err = safe.ReaderGetByID[*omni.Cluster](ctx, client.Omni().State(), talosconfigCmdFlags.cluster)
			if err != nil {
				if !state.IsNotFoundError(err) {
					return err
				}
				
				fmt.Fprintf(os.Stderr, "warning: cluster %q does not exist\n", talosconfigCmdFlags.cluster)
			}
```

1. We have the `safe` package which supports a shorter syntax for the resource operations.
2. More Go way would be to have returns and less nesting/else/else if blocks.
3. `%q` can help to omit explicit quotes and there's no need to inject `\n`
4. The log should go to stderr

Reply:
Thank you for the feedback! I've amended the commit. I also extracted the `getLocalTalosconfigPath` function as the `gocyclo` linter started complaining about the complexity of the `getTalosconfig` function.